ASC-331/ASC-597/Display list of Molecule Orders

VamsiExcelra opened the pull request:
implemented the shared common component for table datagrid with sorting, filtering, row grouping, infinite scroll, columns ordering and options to use across application
implemented the api in route
updated the schema
integrated the api in frontend
handled the internal and external users.

himanshubexcelra commented on apps/web/sharedComponents/table/dataGrid.tsx:
```diff
@@ -0,0 +1,192 @@
+                );
+            }
+
+            if (field === "caco2" || field === "herg" || field === "clint" || field === "hepg2cytox" || field === "solubility") {
```
Columns should come from component calling data grid. It's not fixed. 

Purpose of adding shared component is to use it everywhere in application. It's not onlt for molecule realted data grid.

I want to use it for users and organization table also

VamsiExcelra replied:
changes done and updated

himanshubexcelra commented on apps/web/utils/constants.ts:
```diff
@@ -237,3 +237,105 @@ export const features = [
   { name: "Order Management", value: "order_management", checked: false },
 ];
+
+export const MoleculeOrders = [
```
Fetch molecule order data from api & db.

VamsiExcelra replied:
api implemented to fetch data from db

VamsiExcelra commented:
1. implemented the shared common component for table datagrid with sorting, filtering, row grouping, infinite scroll, columns ordering and options to use across application
2. implemented the api in route
3. updated the schema
4. integrated the api in frontend
5. handled the internal and external users.

himanshubexcelra commented on apps/web/app/molecule_order/page.tsx:
```diff
@@ -0,0 +1,60 @@
+  const isInternalUser = myRoles.some((role: string) => 
+    role === 'researcher' || role === 'protocol_approver');
+  const isExternalUser = myRoles.some((role: string) => 
+    role === 'library_manager' || role === 'admin');
```
admin is not external user org_admin is external user.

VamsiExcelra replied:
updated with user type

himanshubexcelra commented on apps/web/components/MoleculeOrder/MoleculeOrder.tsx:
```diff
@@ -0,0 +1,171 @@
+    },
+  ];
+
+  const handleBookMarkItem = (data: any) => console.log(data, 'data-data');
```
Remove all console log

VamsiExcelra replied:
all consoles removed

himanshubexcelra commented on apps/web/app/molecule_order/page.tsx:
```diff
@@ -0,0 +1,60 @@
+import MoleculeOrderPage from '@/components/MoleculeOrder/MoleculeOrder';
+
+export default async function Page({ searchParams }: 
+  { searchParams: { projectId?: string; libraryId?: string } }) {
```
Define prop type properly. You can check in Header.tsx or LoginForm.tsx.

VamsiExcelra replied:
type added

himanshubexcelra commented on apps/web/app/molecule_order/page.tsx:
```diff
@@ -0,0 +1,60 @@
+      });
+    } else {
+      console.warn("User role not supported or missing required parameters.");
+      return <MoleculeOrderPage initialData={[]} />;
```
There should be only one return statement. You should not decide layout conditionally. You should pass data conditionally to one returned layout.

VamsiExcelra replied:
changed to one return statement

himanshubexcelra commented on apps/web/components/MoleculeOrder/MoleculeOrder.tsx:
```diff
@@ -0,0 +1,171 @@
+  let color: StatusCodeType = 'ready';
+  const value = data[field];
+
+  if (value !== undefined) {
```
This one i am not getting what you are trying to do here.

VamsiExcelra replied:
Dynamic field access based on the `field` parameter

himanshubexcelra commented on apps/web/components/MoleculeOrder/MoleculeOrder.tsx:
```diff
@@ -0,0 +1,171 @@
+  const moleculeColumns = [
+    {
+      dataField: 'bookmark',
+      width: 100,
```
In molecule order this column should not come i guess. I will confirm.

himanshubexcelra commented on apps/web/sharedComponents/table/dataGrid.tsx:
```diff
@@ -0,0 +1,146 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
```
eslint rule is missing here

VamsiExcelra commented on apps/api/app/v1/molecule_order/route.tsx:
```diff
@@ -0,0 +1,65 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
rule removed

VamsiExcelra commented on apps/api/prisma/schema.prisma:
schema updated with main branch schema

VamsiExcelra commented on apps/web/app/molecule_order/page.tsx:
```diff
@@ -0,0 +1,60 @@
+import { redirect } from 'next/navigation';
+import MoleculeOrderPage from '@/components/MoleculeOrder/MoleculeOrder';
+
+export default async function Page({ searchParams }: 
```
page updated 'MoleculeOrder' name

VamsiExcelra commented on apps/web/app/molecule_order/page.tsx:
```diff
@@ -0,0 +1,60 @@
+        libraryId: searchParams.libraryId,
+      });
+    } else {
+      console.warn("User role not supported or missing required parameters.");
```
moved to message.ts

himanshubexcelra commented:
<img width="303" alt="image" src="https://github.com/user-attachments/assets/7c78ee09-4f9e-461f-b273-1daf3fef81cc">

see alignment and spacing is not according to figma screen.

himanshubexcelra commented:
breadcrumb is not active.

himanshubexcelra commented on apps/web/components/MoleculeOrder/MoleculeOrder.tsx:
```diff
@@ -0,0 +1,167 @@
+  molecular_weight: string;
+  smile: string;
+  status: string;
+  yield?: number;
```
these columns are not there in molecule_order.

VamsiExcelra replied:
all the mentioned columns are there in figma design and for 'molecular_weight' i am displaying value too which is coming from 'molecules'

himanshubexcelra commented on apps/web/components/MoleculeOrder/MoleculeOrder.tsx:
```diff
@@ -0,0 +1,167 @@
+  let color: StatusCodeAPIType = 'READY';
+  const value = data[field]; // Dynamic field access based on the `field` parameter
+
+  if (typeof value === 'number') {
```
Status design should be similar to figma and i think this code is not correct.

@ForumExcelra  @VamsiExcelra

VamsiExcelra replied:
color codes are i have checked

himanshubexcelra commented on apps/web/components/MoleculeOrder/MoleculeOrder.tsx:
```diff
@@ -0,0 +1,167 @@
+    },
+  ];
+
+  const handleBookMarkItem = () => { };
```
These all constants not needed. Remove it.

VamsiExcelra replied:
removed

himanshubexcelra commented on apps/web/components/MoleculeOrder/MoleculeOrder.tsx:
```diff
@@ -0,0 +1,167 @@
+      title: 'Status',
+      width: 170,
+      customRender: (data: MoleculeOrder) => {
+        const colorKey = data.status.toUpperCase() as keyof typeof StatusCodeBg;
```
Complete status code looks wrong.

@ForumExcelra  @VamsiExcelra

VamsiExcelra replied:
color codes are i have checked

himanshubexcelra commented:
Send to Synthesis button is not yet added.

himanshubexcelra commented:
As  lib manage i created orders. when i come to molecule orders page i am not able to see orders.
<img width="959" alt="image" src="https://github.com/user-attachments/assets/1d324cba-0e1f-42ff-83ab-439b8e14615e">

There are orders under organization 2 which is fauxbio and for that alos it's empty.

<img width="959" alt="image" src="https://github.com/user-attachments/assets/1d324cba-0e1f-42ff-83ab-439b8e14615e">

himanshubexcelra commented:
<img width="335" alt="image" src="https://github.com/user-attachments/assets/be0bb7fc-dcee-46f3-b04b-8b2707c666e6">
Why external type is added in lowercase.



As per the number of comments increasing . I can say unit testing is not done properly on this PR. Just by assumption you are adding things. 4 story points has reached to 9 story points.

himanshubexcelra commented:
Why here type props is not defined like i explained many times.
<img width="515" alt="image" src="https://github.com/user-attachments/assets/f5d4e2d1-2562-4103-b497-38f3b8b3b033">
Whatever i am explaining it's repeating multiple times and still it's not done.

himanshubexcelra commented:
In shared data grid component why Send for synthesis popup is added.

<img width="581" alt="image" src="https://github.com/user-attachments/assets/af840a64-8779-47bd-bd08-8a3e4745dfe9">

VamsiExcelra commented:
> Send to Synthesis button is not yet added.

send to synthesis button added
